central/group/datastore/internal/store/store_impl.go

```diff
@@ -265,3 +310,37 @@ func getPossibleGroupProperties(authProviderID string, attributes map[string][]s
 	}
 	return
 }
+
+// checkDefaultGroupForProps will check whether the given properties are a default group and, if they are, search the
+// store for the given auth provider ID, checking whether a default group already exists.
+// If the properties do not indicate a default group or the default group does not yet exist, it will return false.
+// Otherwise, it will return true.
+func checkDefaultGroupForProps(tx *bolt.Tx, props *storage.GroupProperties,
+	defaultGroupCache expiringcache.Cache) (bool, error) {
+	// 1. Short-circuit if the props do not indicate a default group. A default group only has the auth provider ID
+	// field set.
+	if !isDefaultGroup(props) {
+		return false, nil
+	}
+	// 2. Short-circuit if the auth providerID is in the cache.
+	if defaultGroupCache.Get(props.GetAuthProviderId()) != nil {
```

Comment:
Is this step for performance reasons? If so, I'd not bother and simplify it away, here is why:
* complexity of a normal check is linear;
* _default_ groups are added and modified rarely, much rarely than groups are mapped to identities;
* linear check should be fine given for group mapping we now have `O(k * n)`.